bytecomp/matching.ml

```diff
@@ -2280,10 +2280,19 @@ let combine_constant loc arg cst partial ctx def
           fail
           (Pbintcomp(Pnativeint, Cneq)) (Pbintcomp(Pnativeint, Clt))
           arg const_lambda_list
```

Comment:
I think it would be more natural (and more readable in the future) to move the `nativeint` case above the `{,u}int{32,64}` case, as in the rest of the changes.